llama_index/postprocessor/rankGPT_rerank.py

```diff
@@ -0,0 +1,223 @@
+import copy
+import logging
+from typing import Any, Dict, List, Optional, Sequence
+
+import tiktoken
+
+from llama_index.bridge.pydantic import Field
+from llama_index.llms import ChatMessage, ChatResponse, OpenAI
+from llama_index.llms.openai_utils import (
+    openai_modelname_to_contextsize,
+    resolve_openai_credentials,
+)
+from llama_index.postprocessor.types import BaseNodePostprocessor
+from llama_index.schema import NodeWithScore, QueryBundle
+from llama_index.service_context import ServiceContext
+
+DEFAULT_OPENAI_MODEL = "gpt-3.5-turbo-0301"
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.WARNING)
+
+
+class RankGPTRerank(BaseNodePostprocessor):
+    """LLM-based reranker."""
+
+    top_n: int = Field(description="Top N nodes to return.")
+    service_context: ServiceContext = Field(
+        description="Service context.", exclude=True
+    )
+    model: str = Field(
+        default=DEFAULT_OPENAI_MODEL, description="The OpenAI model to use."
+    )
+    temperature: float = Field(
+        default=0.0,
+        description="The temperature to use during generation.",
+        gte=0.0,
+        lte=1.0,
+    )
+    api_key: str = Field(default=None, description="The OpenAI API key.", exclude=True)
+    api_base: str = Field(description="The base URL for OpenAI API.")
+    api_version: str = Field(description="The API version for OpenAI API.")
+
+    def __init__(
+        self,
+        service_context: Optional[ServiceContext] = None,
+        top_n: int = 10,
+        model: Optional[str] = "gpt-3.5-turbo",
+        temperature: float = 0.0,
+        api_key: Optional[str] = None,
+        api_base: Optional[str] = None,
+        api_version: Optional[str] = None,
+        **kwargs: Any,
+    ) -> None:
+        service_context = service_context or ServiceContext.from_defaults()
```

Comment:
better to let users pass in an LLM and use that directly, rather than using a service context?